hpedockerplugin/volume_manager.py

```diff
@@ -753,12 +770,19 @@ def _get_snapshot_response(self, snapinfo, snapname):
 
     def _get_snapshot_etcd_record(self, parent_volname, snapname):
         volumeinfo = self._etcd.get_vol_byname(parent_volname)
-
         snapshots = volumeinfo.get('snapshots', None)
+        if 'snap_cpg' in volumeinfo:
+            snapshot_cpg = volumeinfo.get('snap_cpg')
+        else:
+            snapshot_cpg = self._hpeplugin_driver.get_snapcpg(volumeinfo,
+                                                              False)
         if snapshots:
             self._sync_snapshots_from_array(volumeinfo['id'],
-                                            volumeinfo['snapshots'])
+                                            volumeinfo['snapshots'],
+                                            snapshot_cpg)
             snapinfo = self._etcd.get_vol_byname(snapname)
+            snapinfo['snap_cpg'] = snapshot_cpg
```

Comment:
Move this code after the if check. 

getting value before none check is not a correct way.